pkg/controller/seed/reconciler_test.go

```diff
@@ -71,40 +75,68 @@ hjZVcW2ygAvImCAULGph2fqGkNUszl7ycJH/Dntw4wMLSbstUZomqPuIVQ==
 
 		})
 
-		It("Should ensure the correct resources are created", func() {
+		Context("different kubernetes versions", func() {
 
-			resources, err := getResources(
-				serverTLSSecretName,
-				image,
-				cosignPublicKeys,
-				caBundle,
-				failurePolicy,
-				useOnlyImagePullSecrets,
-			)
-
-			Expect(err).ToNot(HaveOccurred())
-			Expect(resources).To(HaveLen(10))
-
-			expectedResources := map[string]string{
-				validatingWebhookKey:  expectedValidatingWebhook(caBundle, failurePolicy),
-				mutatingWebhookKey:    expectedMutatingWebhook(caBundle, failurePolicy),
-				clusterRoleKey:        expectedClusterRole(),
-				clusterRoleBindingKey: expectedClusterRoleBinding(),
-				deploymentKey:         expectedDeployment(namespace, image, cosignSecretName, serverTLSSecretName),
-				pdbKey:                expectedPDB(namespace),
-				cosignSecretNameKey:   expectedSecretCosign(namespace, cosignSecretName, cosignPublicKeys),
-				serviceKey:            expectedService(namespace),
-				serviceAccountKey:     expectedServiceAccount(namespace),
-				vpaKey:                expectedVPA(namespace),
-			}
-
-			for key, expectedResource := range expectedResources {
-				resource, ok := resources[key]
-				Expect(ok).To(BeTrue())
+			JustBeforeEach(func() {
```

Comment:
Please, rework it to be table test and pass the kubernetes version as parameter. Right now it requires multiple before nodes to be properly synced, while a more simple test with one parameter can be used.